ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -81,6 +105,74 @@ const ApproveStaticSimulation = () => {
   );
 };
 
+const SpendingCap = () => {
+  const t = useI18nContext();
+
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const { receivedToken } = useReceivedToken();
+
+  const { tokenAmount, formattedTokenNum, value, pending } =
+    useApproveTokenSimulation(transactionMeta);
+
+  if (pending) {
+    return <Container isLoading />;
+  }
+
+  if (!receivedToken) {
```

Comment:
What happens if someone tries to approve a token they don't own any balance of?

Is that allowed? Can you pre-approve a balance you don't yet have?

Should we show an error?

Will this just omit the spending cap despite it being an approval?

Reply:
> What happens if someone tries to approve a token they don't own any balance of? Is that allowed? Can you pre-approve a balance you don't yet have?

Yes, the approved spending cap is stored in a mapping, so it can be a number higher and independent of the balance (which is stored in another mapping).

```
mapping(address => uint256) balances;
mapping(address => mapping (address => uint256)) allowed;

//...

function approve(address delegate, uint256 numTokens) public override returns (bool) {
    allowed[msg.sender][delegate] = numTokens;
    emit Approval(msg.sender, delegate, numTokens);
    return true;
}
```

> Should we show an error?

I don't think so given the above.

> Will this just omit the spending cap despite it being an approval?

Yes, because we can't establish the spending cap token symbol if it hasn't been added to the wallet first.

The static simulation component uses the contract address (or PetName) instead of the token symbol so it still displays the spending cap in that scenario.